### PR TITLE
infoschema: fix information schema data race

### PR DESCRIPTION
**Pull request by lamxTyler**
## What have you changed? (mandatory)

When applying the changes to `DBInfo`, we should not directly modify it, because it may be referenced by others.

## What is the type of the changes? (mandatory)

- Bug fix (non-breaking change which fixes an issue)

## How has this PR been tested? (mandatory)

Tested in https://github.com/pingcap/tidb/pull/7305
## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No.
## Does this PR affect tidb-ansible update? (mandatory)

No.
## Does this PR need to be added to the release notes? (mandatory)

No.
## Add a few positive/negative examples (optional)

PTAL @coocood @zimulala

### CODE REVIEW

File: infoschema/builder.go [hunk 1]
```diff
@@ -187,31 +189,31 @@ func (b *Builder) applyCreateTable(m *meta.Meta, roDBInfo *model.DBInfo, tableID
 
 	newTbl, ok := b.is.TableByID(tableID)
 	if ok {
-		roDBInfo.Tables = append(roDBInfo.Tables, newTbl.Meta())
+		dbInfo.Tables = append(dbInfo.Tables, newTbl.Meta())
```
Comment (tiancaiamao) [inline infoschema/builder.go]: So, here is the race?

File: infoschema/builder.go [hunk 2]
```diff
@@ -187,31 +189,31 @@ func (b *Builder) applyCreateTable(m *meta.Meta, roDBInfo *model.DBInfo, tableID
...
-	for i, tblInfo := range roDBInfo.Tables {
+	for i, tblInfo := range dbInfo.Tables {
 		if tblInfo.ID == tableID {
-			if i == len(roDBInfo.Tables)-1 {
-				roDBInfo.Tables = roDBInfo.Tables[:i]
+			if i == len(dbInfo.Tables)-1 {
+				dbInfo.Tables = dbInfo.Tables[:i]
 			} else {
-				roDBInfo.Tables = append(roDBInfo.Tables[:i], roDBInfo.Tables[i+1:]...)
+				dbInfo.Tables = append(dbInfo.Tables[:i], dbInfo.Tables[i+1:]...)
```
Comment (tiancaiamao) [inline infoschema/builder.go]: Or here?
In the PR description, you'd better post the data race log, so it's easier for us to understand where is the cause of the problem.

File: infoschema/builder.go (hunk 1 above)
Comment (lamxTyler) [reply infoschema/builder.go]: Here may have the race.

File: infoschema/builder.go (hunk 2 above)
Comment (lamxTyler) [reply infoschema/builder.go]: It actually happens here in the log, but I browse the file and find more places that may have the race.

**Pull request merged**